### PR TITLE
Re-add changes for connect_nodes_nearest

### PR DESCRIPTION
**Pull request by SylviaWhittle**
Let's try this again 😅

### CODE REVIEW

**Comment by SylviaWhittle** [comment]
image

pre-commit checks

File: tests/measure/conftest.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import pytest
+from skimage.morphology import label
+
+from topostats.tracing.nodestats import nodeStats
+
+# pylint: disable=redefined-outer-name
+
+
+@pytest.fixture()
+def network_array_representation_figure_8():
```
Comment (ns-rse) [inline tests/measure/conftest.py:13]: Very minor but could add...

```suggestion
def network_array_representation_figure_8() -> npt.NDArray:
```

..assuming that `import numpy.typing as npt` is added to the imports.

File: topostats/tracing/nodestats.py [hunk 2]
```diff
@@ -3,17 +3,21 @@
 from __future__ import annotations
 
 import logging
-import math
 
 import networkx as nx
 import numpy as np
 import numpy.typing as npt
+from numpy.typing import NDArray
```
Comment (ns-rse) [inline topostats/tracing/nodestats.py:10]: Essentially redundant given line 9.

I'm gradually favouring and getting in the habit of importing at the module level as it makes it easier to know what package modules belong to. In this instance not much of an issue but there may be other cases where there is the same function from different packages and by importing at the module level it makes it easier to distinguish.

A Find and Replace across the file for `NDArray` > `npt.NDArray` and removing this line should solve things.

**Comment by SylviaWhittle** [comment]
Code now runs properly with `topostats process` using a generated default config on `minicircle.spm`

File: topostats/tracing/nodestats.py [hunk 3]
```diff
@@ -140,6 +143,7 @@ def get_node_stats(self) -> tuple:
         LOGGER.info(f"Node Stats - Processing Grain: {self.n_grain}")
         self.conv_skelly = convolve_skeleton(self.skeleton)
         if len(self.conv_skelly[self.conv_skelly == 3]) != 0:  # check if any nodes
+            LOGGER.info(f"[{self.filename}] : Nodestats - {self.n_grain} contains crossings.")
```
Comment (ns-rse) [inline topostats/tracing/nodestats.py:146]: :+1: Great to see more `LOGGER.info()` across everything.

File: topostats/tracing/nodestats.py [hunk 4]
```diff
@@ -682,11 +634,11 @@ def analyse_nodes(self, max_branch_length: float = 20e-9) -> None:
                     for i, angle in enumerate(angles):
                         matched_branches[i]["angles"] = angle
 
-                    """
-                    except ValueError:
-                        LOGGER.error(f"Node {node_no} too complex, see images for details.")
-                        error = True
-                    """
+                    # """
```
Comment (ns-rse) [inline topostats/tracing/nodestats.py:637]: Could this be removed?

File: topostats/tracing/nodestats.py [hunk 5]
```diff
@@ -1053,7 +1008,7 @@ def gaussian(x: npt.NDArray, h: float, mean: float, sigma: float):
 
     def fwhm2(self, heights: npt.NDArray, distances: npt.NDArray, hm: float | None = None) -> tuple:
         """
-        Caculate the FWHM value.
+        Calculate the FWHM value. TODO: Dictionary-ify this one to help saving.
```
Comment (ns-rse) [inline topostats/tracing/nodestats.py:1011]: Recommend capturing this in an issue so we can keep track of it (if not already done).

Comment (SylviaWhittle) [reply topostats/tracing/nodestats.py:1011]: That's a remnant I accidentally left in from Max's notes. removing it.

File: topostats/tracing/nodestats.py (hunk 2 above)
Comment (SylviaWhittle) [reply topostats/tracing/nodestats.py]: Done now, AFAIK I've found all of them.

File: topostats/tracing/nodestats.py (hunk 4 above)
Comment (SylviaWhittle) [reply topostats/tracing/nodestats.py]: 👍

**Pull request merged**